### PR TITLE
Container stats and billing

### LINKED ISSUE #80: Container statistics
Stream app statistics from Docker

### PR DESCRIPTION
**Pull request by anirudhb**
Closes #80, working towards #2

### CODE REVIEW

**Comment by jasonappah** [comment]
Looks good to me so far, I'll be done with hn within the next couple days so we can get that up and running soon

**Review by jasonappah — APPROVED** [review]
Sick! I'm almost done with the HN integration so once this gets merged I'll hook into this

File: pkg/api/apps/delete.go [hunk 1]
```diff
@@ -16,7 +17,18 @@ func handleDELETEApp(c *gin.Context) {
...
+	result := db.DB.Joins("INNER JOIN team_users ON team_users.team_id = apps.team_id").
+		First(&app, "apps.id = ? AND team_users.user_id = ?", id, user.ID)
+	if result.Error != nil {
+		c.JSON(500, gin.H{"status": "error", "message": result.Error})
+		return
+	}
+
+	biller.StopBiller(app.ID)
+
+	result = db.DB.Raw("DELETE FROM apps "+
```
Comment (cjdenio) [inline pkg/api/apps/delete.go:31]: Is it possible to delete records with `Joins()` instead?

File: pkg/biller/biller.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	if err != nil {
+		return err
+	}
+
+	go biller(app, team, stats_stream)
+
+	return nil
+}
+
+var billerOutputs map[uint]map[chan decimal.Decimal]struct{} = make(map[uint]map[chan decimal.Decimal]struct{})
```
Comment (cjdenio) [inline pkg/biller/biller.go:70]: Why use `struct{}` over `interface{}`?

**Review by cjdenio — APPROVED** [review]
lgtm, 2 quick questions

**Comment by cjdenio** [comment]
also, is this supposed to accrue expenses every time the stats stream updates?

File: pkg/api/apps/delete.go (hunk 1 above)
Comment (anirudhb) [reply pkg/api/apps/delete.go:31]: I tried but it didn't work

File: pkg/biller/biller.go (hunk 2 above)
Comment (anirudhb) [reply pkg/biller/biller.go:70]: `struct{}` is zero sized and also we don't need to store any data in the value. This is basically just a set.

**Comment by anirudhb** [comment]
Yes, every time the stats stream updates expenses are accrued. Stats stream updates every second so this is correct behavior.

**Pull request merged**